libbeat/docs/security/basic-auth.asciidoc

```diff
@@ -1,95 +1,148 @@
 [role="xpack"]
 [[beats-basic-auth]]
-=== Configuring authentication credentials for {beatname_uc}
+=== Configure authentication credentials
 
 When sending data to a secured cluster through the `elasticsearch`
 output, {beatname_uc} must either provide basic authentication credentials
 or present a client certificate.
 
 To configure authentication credentials for {beatname_uc}:
 
-. Create a role that has the `manage_index_templates` and
-`monitor` cluster privileges, and `read`, `write`, and `create_index`
-privileges for the indices that {beatname_uc} creates. You can create roles from the
-**Management / Roles** UI in {kib} or through the `role` API.
-For example, the following request creates a ++{beat_default_index_prefix}_writer++ role:
+. Create a writer role that has the following privileges:
 +
+--
+ifeval::["{beatname_lc}"!="filebeat"]
+* *Cluster*: `manage_index_templates` and `monitor`
+endif::[]
+ifeval::["{beatname_lc}"=="filebeat"]
+* *Cluster*: `manage_index_templates`, `monitor`, and
+`manage_ingest_pipelines`
+endif::[]
+* *Index*: `read`, `write`, and `create_index` on the {beatname_uc} indices 
+--
++
+You can create roles from the **Management / Roles** UI in {kib} or through the
+`role` API. For example, the following request creates a role named
+++{beat_default_index_prefix}_writer++:
++
+--
+ifeval::["{beatname_lc}"!="filebeat"]
 ["source","sh",subs="attributes,callouts"]
 ---------------------------------------------------------------
 POST _xpack/security/role/{beat_default_index_prefix}_writer
 {
-  "cluster": ["manage_index_templates", "monitor"],
+  "cluster": ["manage_index_templates","monitor"],
   "indices": [
     {
       "names": [ "{beat_default_index_prefix}-*" ], <1>
-      "privileges": ["write","create_index"]
+      "privileges": ["read","write","create_index"]
```

Comment:
What is the `read` needed for? I can't think of any Beats that would use this. Packetbeat used to a long time ago, but not anymore AFAIK.

Reply:
I don't have the reason in my notes. I'll retest to make sure it's not a copy/paste error.

Reply:
Confirmed that "read" privilege is not required (tested on metricbeat and packetbeat...I'm assuming that's sufficient).